policy/common/trigger.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Trigger is a flag set that marks the types of GitHub events that could
+// change the value of a predicate or evaluation. It is used to optimize
+// evaluation by skipping unnecessary work.
+type Trigger uint32
+
+const (
+	// TriggerStatic is a name for the empty trigger set and means the
+	// computation never needs updating.
+	TriggerStatic Trigger = 0
+
+	TriggerCommit Trigger = 1 << iota
+	TriggerComment
+	TriggerReview
+	TriggerLabel
+	TriggerStatus
+	TriggerPullRequest
```

Comment:
`TriggerPullRequest` is for things like editing the body and title or changing the target branch.